### PR TITLE
Remove core dep

### PR DESCRIPTION
**Pull request by connorwstein**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,12 +1,10 @@
-module github.com/smartcontractkit/solana-integration
+module github.com/smartcontractkit/chainlink-solana
 
 go 1.17
 
 require (
-	github.com/ethereum/go-ethereum v1.10.11
 	github.com/gagliardetto/binary v0.5.0
 	github.com/gagliardetto/solana-go v1.0.2
-	github.com/smartcontractkit/chainlink v1.0.1-0.20211130190746-c1f78221a5e7
```
Comment (RyanRHall) [inline go.mod:9]: :+1:

File: pkg/solana/relay.go [hunk 2]
```diff
@@ -57,6 +58,11 @@ func (r relayer) Healthy() error {
 	return nil
 }
 
+type Transmitter interface {
+	Sign(msg []byte) ([]byte, error)
+	PublicKey() solana.PublicKey
+}
+
```
Comment (RyanRHall) [inline pkg/solana/relay.go:65]: :+1:

**Review by RyanRHall — APPROVED** [review]

**Review by aalu1418 — APPROVED** [review]
LGTM 🚀

File: pkg/solana/relay.go (hunk 2 above)
Comment (krebernisak) [reply pkg/solana/relay.go:65]: Should be `Signer` as it just signs and has an id, doesn't transmit anything.

Np now, we'll tweak things like this at a later time.

File: pkg/solana/relay.go [hunk 3]
```diff
@@ -65,31 +71,25 @@ type OCR2Spec struct {
...
 
-	// on-chain program + 2x state accounts (state + transmissions)
-	ProgramID       solana.PublicKey
-	StateID         solana.PublicKey
-	TransmissionsID solana.PublicKey
+	// on-chain program + 2x state accounts (state + transmissions) + validator program
+	ProgramID          solana.PublicKey
+	StateID            solana.PublicKey
+	ValidatorProgramID solana.PublicKey
+	TransmissionsID    solana.PublicKey
```
Comment (krebernisak) [inline pkg/solana/relay.go:78]: Isn't this available from the `Transmitter` interface as `t.PublicKey()`?

File: pkg/solana/transmitter.go [hunk 4]
```diff
@@ -78,15 +73,17 @@ func (c ContractTracker) Transmit(
...
+		return err
 	}
-	if _, err = tx.Sign(pkGetter); err != nil {
-		return errors.Wrap(err, "error on transaction sign")
+	finalSigBytes, err := c.Transmitter.Sign(msgToSign)
+	if err != nil {
+		return err
 	}
+	var finalSig [64]byte
+	copy(finalSig[:], finalSigBytes)
```
Comment (krebernisak) [inline pkg/solana/transmitter.go:85]: What's the purpose of this final copy?

File: pkg/solana/relay.go (hunk 3 above)
Comment (aalu1418) [reply pkg/solana/relay.go:78]: @krebernisak not sure i understand your question...
`TransmissionsID` refers to the account where the transmissions state is stored...`Transmitter` is the account sending the report tx.

Comment (krebernisak) [reply pkg/solana/relay.go:78]: Ah, yes...

Comment (krebernisak) [reply pkg/solana/relay.go:78]: Was thinking about the `Transmitter` interface name, and then this confused me. Thx!

File: pkg/solana/relay.go (hunk 2 above)
Comment (krebernisak) [reply pkg/solana/relay.go:65]: Cosmos SDK interface example: https://github.com/cosmos/cosmos-sdk/blob/779c7e2e174b1dff4f71329a5662c4b8806d36dd/crypto/keyring/keyring.go#L109

**Pull request merged**